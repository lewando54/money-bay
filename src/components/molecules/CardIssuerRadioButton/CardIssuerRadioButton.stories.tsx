import { StoryObj, Meta } from '@storybook/react-native'
import React from 'react'
import CardIssuerRadioButton from './CardIssuerRadioButton'
import { GOOD_GREEN } from '../../../styling/GlobalTheme.style'
import { useState } from 'react'
import { TCardIssuer } from '../../../utils/cardIssuers'

interface ICardIssuerRadioButtonMockupProps {
	cardIssuer: TCardIssuer
	chosen: boolean
    color?: string
	onPress: () => void
}

function StatefulCardIssuerRadioButton({
    cardIssuer='visa-color',
    chosen=false,
    color=GOOD_GREEN,
    onPress
}: ICardIssuerRadioButtonMockupProps){
    const [isChosen, setIsChosen] = useState(chosen)

    return (
        <CardIssuerRadioButton cardIssuer={cardIssuer} chosen={isChosen} color={color} onPress={(id, group, chosen) => {setIsChosen(!chosen); onPress()}}/>
    )
}

const meta = {
    component: StatefulCardIssuerRadioButton,
    argTypes: {
        cardIssuer: { control: 'select', options: [ 'mastercard', 'visa-color' ]},
        chosen: { control: 'boolean' },
        color: { control: 'color' },
        onPress: { action: 'Pressed' }
    }
} satisfies Meta<typeof StatefulCardIssuerRadioButton>

export default meta
type Story = StoryObj<typeof StatefulCardIssuerRadioButton>

export const Default: Story = {
    args: {
        cardIssuer: 'visa-color',
        chosen: false,
        color: GOOD_GREEN
    }
}

export const Mastercard: Story = {
    args: {
        cardIssuer: 'mastercard',
        chosen: false,
        color: GOOD_GREEN
    }
}

export const Chosen: Story = {
    args: {
        cardIssuer: 'visa-color',
        chosen: true,
        color: GOOD_GREEN
    }
}

export const ChosenCustomColor: Story = {
    args: {
        cardIssuer: 'visa-color',
        chosen: true,
        color: '#f00'
    }
}

