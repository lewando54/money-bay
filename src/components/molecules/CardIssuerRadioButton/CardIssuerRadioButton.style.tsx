import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    mainContainer: {
        flexShrink: 1,
        flexGrow: 0,
        alignItems: 'flex-start'
    },
    univContainer: {
        flexDirection: 'row',
        backgroundColor: 'white',
        flexShrink: 1,
        flexGrow: 0,
        justifyContent: 'flex-start',
        alignItems: 'center',
        gap: 10,
        width: 'auto',
        paddingHorizontal: 20
    },
    image: {
        resizeMode: 'contain',
        width: 40,
        maxHeight: 16
    }
})
