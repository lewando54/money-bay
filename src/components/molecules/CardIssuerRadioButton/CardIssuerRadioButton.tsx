import { Image, View } from 'react-native'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import cardIssuerRadioButtonStyle from './CardIssuerRadioButton.style'
import RadioButton, { IRadioButtonProps } from '../../atoms/RadioButton/RadioButton'
import { TCardIssuer, getIssuerLogo } from '../../../utils/cardIssuers'
import { GOOD_GREEN } from '../../../styling/GlobalTheme.style'

interface ICardIssuerRadioButtonProps extends IRadioButtonProps {
    cardIssuer?: TCardIssuer
}

export default function CardIssuerRadioButton({
    cardIssuer='visa-color',
    chosen=false,
    color=GOOD_GREEN,
    id=0,
    group='default',
    onPress,
    testID
}: ICardIssuerRadioButtonProps){
    return (
        <View style={{flexShrink: 1, flexGrow: 0, alignItems: 'flex-start'}}>
            <UniversalContainer testID={testID} style={{flexDirection: 'row', backgroundColor: 'white', flexShrink: 1, flexGrow: 0, justifyContent: 'flex-start', alignItems: 'center', gap: 10, width: 'auto', paddingHorizontal: 20}}>
                <RadioButton chosen={chosen} id={id} group={group} onPress={onPress} color={color}/>
                <Image style={{resizeMode: 'contain', width: 40, maxHeight: 16}} source={getIssuerLogo(cardIssuer)}/>
            </UniversalContainer>
        </View>
    )
}
