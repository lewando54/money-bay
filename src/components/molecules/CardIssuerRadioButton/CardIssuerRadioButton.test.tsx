import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import CardIssuerRadioButton from './CardIssuerRadioButton'
describe('CardIssuerRadioButton', () => {
  jest.useFakeTimers();
  it('should render correctly', () => {
    const tree = renderer.create(<CardIssuerRadioButton chosen={false} onPress={jest.fn()}/>)
    expect(tree).toMatchSnapshot()
  })
})
