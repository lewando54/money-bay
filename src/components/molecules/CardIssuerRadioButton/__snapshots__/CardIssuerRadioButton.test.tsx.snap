// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardIssuerRadioButton should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "flexGrow": 0,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": "auto",
          "padding": 10,
          "width": "100%",
        },
        false,
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "flexGrow": 0,
          "flexShrink": 1,
          "gap": 10,
          "justifyContent": "flex-start",
          "paddingHorizontal": 20,
          "width": "auto",
        },
      ]
    }
    testID="universal-container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#CED6E1",
          "borderRadius": 100,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    >
      <View
        animatedStyle={
          {
            "value": {
              "height": 0,
              "width": 0,
            },
          }
        }
        collapsable={false}
        style={
          {
            "backgroundColor": "#3EB290",
            "borderRadius": 100,
            "height": 0,
            "width": 0,
          }
        }
      />
    </View>
    <Image
      source={1}
      style={
        {
          "maxHeight": 16,
          "resizeMode": "contain",
          "width": 40,
        }
      }
    />
  </View>
</View>
`;
